src/nvim/fold.c

```diff
@@ -788,6 +788,22 @@ void foldUpdate(win_T *wp, linenr_T top, linenr_T bot)
   }
 }
 
+// foldUpdateInsert()
+//
+// Update folds for the insert changes in the buffer of a window.
+//
+void foldUpdateInsert(void)
+{
+  if (foldmethodIsManual(curwin)
+      || foldmethodIsSyntax(curwin)
+      || foldmethodIsExpr(curwin)) {
+    return;
```

Comment:
I think them should be updated.
manual:  No need to update
syntax and expr: Auto update is slow.  It should be updated manually like FastFold plugin